meteor/client/ui/RundownView/RundownTiming/RundownTimingProvider.tsx

```diff
@@ -412,7 +419,22 @@ export const RundownTimingProvider = withTracker<
 					this.partDisplayDurations[partInstancePartId] = partDisplayDuration
 					this.partDisplayDurationsNoPlayback[partInstancePartId] = partDisplayDurationNoPlayback
 					startsAtAccumulator += this.partDurations[partInstancePartId]
-					displayStartsAtAccumulator += this.partDisplayDurations[partInstancePartId] // || this.props.defaultDuration || 3000
+
+					if (playlist.previousPartInstanceId !== partInstance._id) {
+						displayStartsAtAccumulator += this.partDisplayDurations[partInstancePartId]
+					} else {
+						if (this.prevPartId !== unprotectString(playlist.previousPartInstanceId)) {
+							this.lastTakeAt = now
+							this.prevPartId = unprotectString(playlist.previousPartInstanceId) || ''
```

Comment:
I didn't notice this one before, but if you change `prevPartId` you can remove the `unprotectString` here as well and you'll end up with a nice `this.prevPartId = playlist.previousPartInstanceId`, since `playlist.previousPartInstanceId` is also a `PartInstanceId | null` type.